api/v1beta1/servicel2status_types.go

```diff
@@ -26,6 +26,10 @@ import (
 type MetalLBServiceL2Status struct {
 	// Node indicates the node that receives the directed traffic
 	Node string `json:"node,omitempty"`
+	// ServiceName indicates the service this status represents
+	ServiceName string `json:"serviceName,omitempty"`
+	// ServiceNamespace indicates the namespace of the service
+	ServiceNamespace string `json:"serviceNamespace,omitempty"`
```

Comment:
let's make these fields immutable:
```go
	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
	Node string `json:"node,omitempty"`
	// ServiceName indicates the service this status represents
	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
	ServiceName string `json:"serviceName,omitempty"`
	// ServiceNamespace indicates the namespace of the service
	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
	ServiceNamespace string `json:"serviceNamespace,omitempty"`
```
verified on a local cluster that it actually works when using:
```
k edit -n metallb-system servicel2status ".." --subresource=status
```